packages/react-devtools-extensions/src/main/fetchFileWithCaching.js

```diff
@@ -107,17 +107,36 @@ const fetchFromPage = async (url, resolve, reject) => {
   });
 };
 
-// Fetching files from the extension won't make use of the network cache
-// for resources that have already been loaded by the page.
-// This helper function allows the extension to request files to be fetched
-// by the content script (running in the page) to increase the likelihood of a cache hit.
-const fetchFileWithCaching = url => {
+// 1. Check if resource is available via chrome.devtools.inspectedWindow.getResources
+// 2. Check if resource was loaded previously and available in network cache via chrome.devtools.network.getHAR
+// 3. Fallback to fetching directly from the page context (from backend)
+async function fetchFileWithCaching(url: string): Promise<string> {
+  if (__IS_CHROME__ || __IS_EDGE__) {
+    const resources = await new Promise(resolve =>
+      chrome.devtools.inspectedWindow.getResources(r => resolve(r)),
+    );
+
+    // This is a hacky way to make it work for Next, since their URLs are not normalized
+    const normalizedReferenceURL = url.replace('/./', '/');
+    const resource = resources.find(r => r.url === normalizedReferenceURL);
```

Comment:
We can't because this event is emitted once resource is added: if main frame resources are loaded and user then opens browser DevTools, the listener won't be called with main frame resources.

If this gets slow for larger apps, I can work on some caching and normalization (to get resource in O(1)), probably